.github/workflows/build.yml

```diff
@@ -2,6 +2,34 @@ name: Build
 
 on: [push]
 jobs:
+  tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+      
+      - name: Install Base Dependencies
+        run: ${{github.workspace}}/build-scripts/for-linux/prepare-debian-based.sh
```

Comment:
You want to say add a 'test' parameter to the prepare-debian-based.sh to keep one script only ?

Reply:
No, prepare-debian-for-tests.sh is good.

I want the warkflow would call prepare-debian-for-tests.sh, and prepare-debian-for-tests.sh would call prepare-debian-based.sh itself.

Reply:
Please remove these two lines.

Reply:
You still call `prepare-debian-based.sh` from the workflow directly. It is not necessary because it is also called from  `prepare-debian-for-tests.sh`

Reply:
You haven't still removed the lines 13 and 14